src/arrayMethodSort.js

```diff
@@ -4,9 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = noArguments) {
+    let count = 0;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        const prev = this[i - 1];
+        const current = this[i];
+
+        if (compareFunction(prev, current) > 0) {
+          this[i - 1] = current;
+          this[i] = prev;
+          count++;
+        }
+      }
+    } while (count > 0);
+
+    return this;
   };
 }
 
+const noArguments = (a, b) => a.toString() > b.toString();
+
```

Comment:
I think that using external variables in a function is a bad practice.
```suggestion
  [].__proto__.sort2 = function(compareFunction = (a, b) => a.toString() > b.toString()) {
    let count = 0;

    do {
      count = 0;

      for (let i = 1; i < this.length; i++) {
        const prev = this[i - 1];
        const current = this[i];

        if (compareFunction(prev, current) > 0) {
          this[i - 1] = current;
          this[i] = prev;
          count++;
        }
      }
    } while (count > 0);

    return this;
  };
}



```